pyboy/core/cartridge/base_mbc.py

```diff
@@ -103,7 +105,55 @@ def init_rambanks(self, n):
         self.rambanks = memoryview(array.array("B", [0] * (8*1024*16))).cast("B", shape=(16, 8 * 1024))
 
     def getgamename(self, rombanks):
-        return "".join([chr(rombanks[0, x]) for x in range(0x0134, 0x0142)]).split("\0")[0]
+        return "".join([chr(rombanks[0, x]) for x in range(0x0134, 0x0143)]).split("\0")[0]
+
+    def getgametype(self, rombanks):
+        if rombanks[0, 0x143] == 0x80 or rombanks[0, 0x143] == 0xc0:
+            return "Game Boy Color"
+        elif rombanks[0, 0x146] == 0x03:
+            return "Super Game Boy"
+        else:
+            return "Game Boy"
```

Comment:
It's probably better to split this out into separate "cgb" and "sgb" checks, as they are seemingly not mutually exclusive https://gbdev.io/pandocs/The_Cartridge_Header.html